feat(shortcode): add shortcode for asciinema

Lamber-maybe opened the pull request:
Hi @imfing,

Thank you so much for your encouragement on issue #776 As you suggested, I've submitted this pull request with an initial implementation to bring this feature to Hextra.

This PR introduces support for embedding Asciinema players via a new shortcode.

### Key Changes
* **New Dependencies:** Added `assets/css/asciinema-player.css` and `assets/js/asciinema-player.min.js` from the official Asciinema Player repository ([https://github.com/asciinema/asciinema-player](https://github.com/asciinema/asciinema-player)).
* **Global Includes:** Modified `layouts/_partials/head.html` to make these assets available on all pages.
* **Initialization Script:** Created a new file, `assets/js/asciinema-init.js`, to handle the initialization of the Asciinema players.
* **Demo Content:** Created a new page `exampleSite/content/blog/asciinema-demo-en.md` to demonstrate the usage and shortcode.

### Known Issue & Request for Help
While the core functionality is working, there is a minor CSS issue: **the play button is not correctly centered in the player.**

I would greatly appreciate some help with this alignment problem. A key consideration is that **I want to avoid modifying the official `asciinema-player.css` file directly** to ensure that future updates remain straightforward and maintainable. I'm not very strong with CSS and would greatly appreciate it if someone could help resolve this alignment problem.

I'm happy to make any other changes you suggest. Looking forward to your review and feedback!

Lamber-maybe commented:
Hi @imfing ,

Thanks for the review! I've addressed all the comments and resolved the conversations. Let me know if there's anything else.

imfing commented on exampleSite/content/docs/guide/shortcodes/_index.md:
```diff
@@ -17,6 +17,7 @@ Hextra provides a collection of beautiful shortcodes to enhance your content.
   {{< card link="icon" title="Icon" icon="badge-check" >}}
   {{< card link="steps" title="Steps" icon="one" >}}
   {{< card link="tabs" title="Tabs" icon="collection" >}}
+  {{< card link="asciinema" title="Asciinema Player" icon="terminal" >}}
```
for now let's move it to the section below under "Additional shortcodes provided by Hugo and Hextra:"

imfing commented on exampleSite/content/docs/guide/shortcodes/asciinema.md:
```diff
@@ -0,0 +1,65 @@
+---
```
For now, let's exclude it from the sidebar (similar to the jupyter). We'll need to think about better grouping for the sidebar for these shortcodes

imfing commented on static/casts/advanced-demo.cast:
```diff
@@ -0,0 +1,61 @@
+{"version": 2, "width": 80, "height": 24, "timestamp": 1640995200, "env": {"TERM": "xterm-256color", "SHELL": "/bin/bash"}, "title": "Advanced Terminal Demo"}
```
is this used anywhere?

imfing reviewed (commented):
The implementation and documentation are solid.

Let's refine the documentation slightly before merging.

Lamber-maybe commented:
Thanks for the feedback. I've fixed the play button centering issue and updated the documentation as well. Could you please take another look and let me know if there's anything else that needs to be improved?

imfing commented on layouts/_shortcodes/asciinema.html:
```diff
@@ -0,0 +1,48 @@
+{{- /* Get parameters */ -}}
+{{- $castFile := .Get "file" | default (.Get 0) -}}
+{{- /* Handle file path: if not absolute path or URL, prepend /casts/ */ -}}
+{{- if and (not (hasPrefix $castFile "/")) (not (hasPrefix $castFile "http://")) (not (hasPrefix $castFile "https://")) -}}
+  {{- $castFile = printf "/casts/%s" $castFile -}}
+{{- end -}}
```
I noticed that it prepends `/casts/` to paths, which might be confusing to users, so I recommend we raise an error instead of attempting to force the location.

Lamber-maybe commented:
@imfing Thank you for your encouragement and thorough review! 

This was my first open source contribution, and your guidance helped me take that first step. Before this, I could only submit issues and wait for others to implement them. But now, thanks to your encouragement, I feel motivated to create PRs myself when I have good ideas.

Your detailed feedback throughout the review process was incredibly helpful and made this collaboration truly rewarding. Thanks again! 🎉

imfing commented:
@Lamber-maybe keep up the good work 💪
